rules/prefer-type-error.js

```diff
@@ -0,0 +1,127 @@
+'use strict';
+
+const tcIdentifiers = new Set([
+	'isArguments',
+	'isArray',
+	'isArrayBuffer',
+	'isArrayLike',
+	'isArrayLikeObject',
+	'isBoolean',
+	'isBuffer',
+	'isDate',
+	'isElement',
+	'isEmptyObject',
+	'isError',
+	'isFinite',
```

Comment:
Hmm.. [ES treats +/-Infinity as special Number type](https://tc39.github.io/ecma262/#sec-ecmascript-language-types-number-type) (like NaN). So from a technical perspective i'd say it is a `TypeError` like NaN.

From a users perspective this might not be intuitive in a semantical way, especially if you compare this meaning to the infinity of math. Where in my opinion infinity is mostly used to create a line where the number's values aren't of interest anymore. Whereas ES creates the line where numbers can't be treated anymore. So i personally take that as a sign of an invalid result of a math operation.